### DISCUSSION TITLE
Using I18N and primefaces/primelocale

### DISCUSSION THREAD

**Discussion by Cyril30** [primefaces/community#633] [PrimeVue] [answered]
Hello, 

We are using i18n in our PrimeVue application, and trying to add the primelocale for several countries.
I've seen one (great) usage from https://github.com/orgs/primefaces/discussions/478 , but without i18n.

Currently we have this : 

```
//Translation for app texts
const i18n = createI18n({
    locale: navigator.language.split('-')[0],
    //locale: 'fr',
    fallbackLocale: 'en',
    messages: {
        fr: (await import('../i18n/fr.json')).default,
        en: (await import('../i18n/en.json')).default,
    },
});
//Translation for  PrimeVue
import translations from '../i18n/fr.js';
app.use(PrimeVue, {
    ripple: true,
    locale: translations,
});

app.use(i18n);
```

And an example for the Calendar component, we have to put specific data inside :

`<Calendar id="date" v-model="date" dateFormat="dd/mm/yy" touchUI showButtonBar showIcon iconDisplay="input"
                :monthNames="$t('monthNames')" :dayNames="$t('dayNamesShort')" />`

The question is : Is there a different way to implement the primelocale files using i18n to avoid specifiying explicitely  :monthNames="$t('monthNames')" :dayNames="$t('dayNamesShort') ?

Thank you very much and have a nice day :)

Cyril

**Comment by tugcekucukoglu** [comment, marked as answer]
Hello,

There is a starter [i18n example](https://github.com/primefaces/primevue-examples/tree/main/vue-i18n). Also `monthNames` and `dayNames` are not valid properties. And PrimeVue offers its own [Locale API](https://primevue.org/configuration/#locale)

**Reply by Cyril30** [reply]
Thank you very much ! 
I was able to integrate this usage. 
I still have to work to change the dropdown values by flags for example that could use each locale code, but it is a good start ! Thanks again

**Reply by BenJackGill** [reply]
How did you get this working? That i18n example is now missing (404 error)

**Reply by jinynet** [reply]
see [Locale API](https://primevue.org/configuration/#locale)

**Reply by melloware** [reply]
Here is a runnable example showing you how to use PrimeLocale with PrimeVue: https://codesandbox.io/p/devbox/primevue-locale-loader-forked-mxkhh8?file=%2Fsrc%2FApp.vue

**Reply by Mo-shi** [reply]
Hi can you send other. its gone i cant see it. Sor my en is bad.

**Reply by Mo-shi** [reply]
the example

**Reply by Mo-shi** [reply]
i want  use CN. But i dont know how to use.

**Reply by melloware** [reply]
@Mo-shi try this one: https://codesandbox.io/p/devbox/primevue-locale-loader-forked-mxkhh8?file=%2Fsrc%2FApp.vue

**Reply by Mo-shi** [reply]
the first link is gone.
thank you.

**Reply by Mo-shi** [reply]
@melloware  thank you. got it.
